src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -516,6 +520,23 @@ public FileChannel getChannel() {
         return fc;
     }
 
+    @SuppressWarnings("unused")
+    private final JDKFileResource resource = new JDKFileResource(this) {
+        @Override
+        protected FileDescriptor getFD() {
+            return fd;
+        }
+
+        @Override
+        protected Supplier<Exception> claimException(int fdNum, String path) {
+            if (fd == FileDescriptor.in) {
+                return null;
+            } else {
+                return super.claimException(fdNum, path);
+            }
+        }
+    };
+
```

Comment:
Providing full range of policies for FileInputStreams seems inherently unsafe, e.g. *_AT_END. I also think we need to distinguish FileInputStreams created with a path or with fd.